### PR TITLE
Cleanup of Sif{0,1,2}.cpp

### PR DESCRIPTION
**Pull request by xsacha**
Some unused functions.
Local variable 'done' wasn't being used.
Collapse it in to main functions so that it is easier to (re)move.

Functionality is unchanged, although it appears most of it isn't used or is only ever used once.

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
Are you still working on this PR or can we abandon this?

**Comment by xsacha** [comment]
PR was finished                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        From: refractionpcsx2Sent: Sunday, 27 September 2015 4:53 AMTo: PCSX2/pcsx2Reply To: PCSX2/pcsx2Cc: xsachaSubject: Re: [pcsx2] Cleanup of Sif{0,1,2}.cpp (#270)Are you still working on this PR or can we abandon this?

—Reply to this email directly or view it on GitHub.

**Comment by refractionpcsx2** [comment]
Ok didn't know where you'd gotten to as you hadn't said you were finished :P I will try this when I get chance, although it looks like you've just made it how it was before things were put in to functions, so this is essentially a revert :P

**Comment by ssakash** [comment]
@xsacha 

https://github.com/PCSX2/pcsx2/commit/31ee576d6a0f015199d91948602a79bcc64c2d22 caused a merge conflict on the PR.  blame ref :p 

Personally I think the commit doesn't make the code more readable, it just makes it harder to review and find potential important commits on history. would be way better if it only dealt with unused variables and only made some clean up on important and necessary parts.

**Comment by refractionpcsx2** [comment]
As i said, this is essentially a revert to how it used to be, i think arcum or someone completely rewrote it to be how it is now ages ago.

**Comment by xsacha** [comment]
Just over a year later guys...

File: pcsx2/sif2.cpp [hunk 1]
```diff
@@ -339,13 +238,18 @@ __fi void SIF2Dma()
...
-				BusyCheck++;
+				busyCheck = true;
 				HandleEETransfer();
 			}
 		}
-	} while (/*!done && */BusyCheck > 0); // Substituting (sif2.ee.busy || sif2.iop.busy) breaks things.
+		if (!busyCheck)
+			break;
+	}
+
```
Comment (xsacha) [inline pcsx2/sif2.cpp]: No, it is above.
Code works fine.

**Comment by ssakash** [comment]
@xsacha 

There are lots of non-functional changes which could be done on the code base,. I really agree with @ramapcsx2 here.  It would make more sense to clean them up when making any change at the functions.

**Comment by xsacha** [comment]
Except that no one will pull it because they don't understand the changes if you are also cleaning it up.

**Comment by ssakash** [comment]
> Except that no one will pull it because they don't understand the changes if you are also cleaning it up.

I'm pretty sure that @refractionpcsx2 and @sudonim1 do understand the SIF parts of the code. while I have pretty much been trying to understand lots of the SIF parts but only have managed to partially understand the stall control stuffs and priority tags :p 

I do understand your concern though, would be nice if you could contribute in the future though. Thanks for the effort ! 

also if someone doesn't understand one of my changes, I'll try to explain them. ;)

**Comment by refractionpcsx2** [comment]
It isn't just a simple clean up though, I'm reluctant to pull it because it is a step backwards on a change that was made ages ago, a lot was functioned to make it more versatile in the future and easier to read instead of being massive blocks of text, you reversed that.

I just personally feel it wasn't a step forward.

**Comment by xsacha** [comment]
I don't see how having single-use functions benefits readability. It means you have to keep looking in multiple places to read the code.
In any case, there are twice as many deletions as insertions with every used line and comment maintained AFAIK. Not sure how that can be explained as massive blocks of text.

**Comment by ramapcsx2** [comment]
This is how complex problems like this are traditionally solved in PCSX2:
Someone recognizes problem code. They start researching the component using all available documents. Next they will discuss their understanding on IRC while planning a rewrite. Next there'll be some rather big commits: a total rewrite of the system that behaves functionally the same (or better) but does it far more efficient.
This code needs someone like that. It should not just be tweaked and condensed, it should be re-designed and rewritten completely.
This requires spot on knowledge of the 3 SIF channels and the IOP / EE hooks, and additionally, it'd be great to conserve cycles here, as SIF is a high speed interface.

**Pull request closed without merging**